### PR TITLE
Add openstack-baremetal-operator container image build job

### PR DESCRIPTION
**Pull request by marios**
This adds a image build job for the openstack-baremetal-operator As discussed at [1] we cannot have a build-deploy job so this can at least provide some coverage for the image build.

[1] https://github.com/openshift/release/pull/37291#issuecomment-1471907258

### CODE REVIEW

**Comment by marios** [comment]
@viroel @Sandeepyadav93 grateful for initial sanity check when you next have time for reviews please o/ thank you!

File: ci-operator/config/openstack-k8s-operators/openstack-baremetal-operator/openstack-k8s-operators-openstack-baremetal-operator-master.yaml [hunk 1]
```diff
@@ -6,6 +15,22 @@ images:
...
+  inputs:
+    cli:
+      paths:
+      - destination_dir: .
+        source_path: /usr/bin/oc
+    opm-builder:
+      paths:
+      - destination_dir: .
+        source_path: /bin/opm
+  to: openstack-operator-ci-build-deploy
```
[the remark this answers is not in the document]
Comment (Sandeepyadav93) [reply ci-operator/config/openstack-k8s-operators/openstack-baremetal-operator/openstack-k8s-operators-openstack-baremetal-operator-master.yaml:33]: @marios we need to keep it as here we create an image openstack-operator-ci-build-deploy which we use in build/deploy/kuttl steps .

https://github.com/openshift/release/blob/master/ci-operator/step-registry/openstack-k8s-operators/build/openstack-k8s-operators-build-ref.yaml#L3

**Comment by Sandeepyadav93** [comment]
/pj-rehearse ack

We are only adding container build job for now, results looks good
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37492/rehearse-37492-pull-ci-openstack-k8s-operators-openstack-baremetal-operator-master-openstack-baremetal-operator-build/1638198248692781056/artifacts/openstack-baremetal-operator-build/openstack-k8s-operators-build/build-log.txt

**Comment by Sandeepyadav93** [comment]
/lgtm

I am not in owners file, so will need vote from someone in owners file.

**Review by abays — APPROVED** [review]
/lgtm

File: ci-operator/config/openstack-k8s-operators/openstack-baremetal-operator/openstack-k8s-operators-openstack-baremetal-operator-master.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (marios) [reply ci-operator/config/openstack-k8s-operators/openstack-baremetal-operator/openstack-k8s-operators-openstack-baremetal-operator-master.yaml:33]: thank you for checking @Sandeepyadav93++

**Pull request merged**